src/encryption/encryption.ts

```diff
@@ -184,46 +131,48 @@ export async function encryptWithKeyDerivedFromString(
  * key `encryptWithGeneratedKey`.
  */
 
+/**
+ * @deprecated This method should be replaced by
+ * encryptWithKey method. This method convert give data string to UTF-8 bytes utf8
+ * before enctryptin it
+ */
 export async function encryptStringWithKey(
-  { key, data, strategy, iv }: IEncryptionOptions,
+  key: string,
+  data: string,
+  strategy: CipherStrategy,
+  iv?: string,
   serializationVersion: SerializationFormat = SerializationFormat.latest_version
 ): Promise<IEncryptionResult> {
-  return encryptWithKey({ key, data, strategy, iv }, serializationVersion, 'utf8');
+  if (!data || data === '') {
+    return { encrypted: null, serialized: null };
+  }
+  return encryptWithKey({ key, data: utf8ToBytes(data), strategy, iv }, serializationVersion);
 }
 
+/**
+ * @deprecated This method should be replaced by
+ * encryptWithKey method. This method convert give data to binary bytes
+ */
 export async function encryptBinaryWithKey(
-  { key, data, strategy, iv }: IEncryptionOptions,
+  key: string,
+  data: string,
+  strategy: CipherStrategy,
+  iv?: string,
   serializationVersion: SerializationFormat = SerializationFormat.latest_version
 ): Promise<IEncryptionResult> {
-  return encryptWithKey({ key, data, strategy, iv }, serializationVersion, 'raw');
+  if (!data || data === '') {
+    return { encrypted: null, serialized: null };
+  }
+  return encryptWithKey({ key, data: binaryToBytes(data), strategy, iv }, serializationVersion);
```

Comment:
`decodeByteString` here - or just use the existing `stringAsBinaryBuffer` method that exists in the codebase.